src/foundry/client-esm/dice/roll.d.mts

```diff
@@ -0,0 +1,636 @@
+import type { ConfiguredDocumentClass } from "../../../types/helperTypes.d.mts";
+import type { ConstructorOf, DeepPartial, InexactPartial } from "../../../types/utils.d.mts";
+
+import type { RollParseNode } from "./_types.d.mts";
+import type DiceTerm from "./terms/dice.d.mts";
+import type PoolTerm from "./terms/pool.d.mts";
+import type RollTerm from "./terms/term.d.mts";
+import type RollResolver from "../applications/dice/roll-resolver.d.mts";
+
+/**
+ * An interface and API for constructing and evaluating dice rolls.
+ * The basic structure for a dice roll is a string formula and an object of data against which to parse it.
+ *
+ * @typeParam D - the type of data object against which to parse attributes within the formula
+ *
+ * @example Attack with advantage!
+ * ```typescript
+ * let r = new Roll("2d20kh + @prof + @strMod", {prof: 2, strMod: 4});
+ *
+ * // The parsed terms of the roll formula
+ * console.log(r.terms);    // [Die, OperatorTerm, NumericTerm, OperatorTerm, NumericTerm]
+ *
+ * // Execute the roll
+ * await r.evaluate();
+ *
+ * // The resulting equation after it was rolled
+ * console.log(r.result);   // 16 + 2 + 4
+ *
+ * // The total resulting from the roll
+ * console.log(r.total);    // 22
+ * ```
+ */
+declare class Roll<D extends object = {}> {
```

Comment:
1. `object` allows any function, array, or object. You probably want `Record<string, unknown>` which is only objects.
2. `{}` as a type allows anything except null and undefined. You probably want `Record<string, never>` to represent a type with no properties.